radicle-remote-helper/src/lib.rs

```diff
@@ -0,0 +1,154 @@
+#![allow(clippy::collapsible_if)]
+use std::path::PathBuf;
+use std::{env, io, process};
+
+use radicle::crypto::{PublicKey, Signer};
+use radicle::node::Handle;
+use radicle::ssh;
+use radicle::storage::{ReadRepository, WriteStorage};
+
+/// The service invoked by git on the remote repository, during a push.
+const GIT_RECEIVE_PACK: &str = "git-receive-pack";
+
+#[derive(Debug)]
+pub enum Error {
+    /// Remote repository not found (or empty).
+    RepositoryNotFound(PathBuf),
+    /// Secret key is not registered, eg. with ssh-agent.
+    KeyNotRegistered(PublicKey),
+    /// Invalid command received.
+    InvalidCommand(String),
+    /// Invalid arguments received.
+    InvalidArguments(Vec<String>),
+    /// Invalid git remote URL.
+    InvalidRemoteUrl(radicle::git::Url),
+    /// Invalid remote repository identifier.
+    InvalidRemoteId(radicle::git::Url, radicle::identity::IdError),
+}
+
+impl std::error::Error for Error {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        None
+    }
+}
```

Comment:
Pretty sure you can just do the below because of default implementations:
```suggestion
impl std::error::Error for Error {}
```